stoken/main.py

```diff
@@ -48,8 +49,15 @@ def main(mode: str, encoding: str, variable_prefix: str, variable_suffix: str, d
         except Exception as e:
             quit_with_info(f"{Fore.RED}{e}\n\n"
                            f"Fail to read configuration file. Please refer to {README_URL} and modify the `{YAML_PATH}`.{colorama.Style.RESET_ALL}")
-
+    git_dev = BASE_DIR.resolve()
+    while not list(git_dev.glob("*.git")):
+        git_dev = git_dev.parent
+    print(git_dev)
+    repo = git.repo.base.Repo(git_dev)
     for file in BASE_DIR.glob("**/*"):
+        if repo.ignored(file):
```

Comment:
需要考虑 repo == None 的情况

```suggestion
        if repo and repo.ignored(file):
```